File: courtlistener/api/investments.py
"""
Investments API module for CourtListener SDK.
"""

from typing import Dict, Any, Optional, List
from .base import BaseAPI


class InvestmentsAPI(BaseAPI):
    """API for accessing investment data from financial disclosures."""
    
    def __init__(self, client):
        super().__init__(client)
        self.endpoint = "investments/"
    
    def list(
        self,
        financial_disclosure: Optional[int] = None,
        judge: Optional[int] = None,
        **kwargs
    ) -> Dict[str, Any]:
        """
        List investment records.
        
        Args:
            financial_disclosure: Filter by financial disclosure ID
            judge: Filter by judge ID
            **kwargs: Additional query parameters
        
        Returns:
            API response with paginated list of investments
        """
        params = {}
        if financial_disclosure is not None:
            params['financial_disclosure'] = financial_disclosure
        if judge is not None:
            params['judge'] = judge
        
        params.update(kwargs)
        return self.client.get(self.endpoint, params=params)
    
    def get(self, investment_id: int) -> Dict[str, Any]:
        """
        Get a specific investment by ID.
        
        Args:
            investment_id: Investment ID
        
        Returns:
            Investment data
        """
        return self.client.get(f"{self.endpoint}{investment_id}/")
    
    def paginate(
        self,
        financial_disclosure: Optional[int] = None,
        judge: Optional[int] = None,
        **kwargs
    ):
        """
        Get paginated investments.
        
        Args:
            financial_disclosure: Filter by financial disclosure ID
            judge: Filter by judge ID
            **kwargs: Additional query parameters
        
        Returns:
            PageIterator for investments
        """
        params = {}
        if financial_disclosure is not None:
            params['financial_disclosure'] = financial_disclosure
        if judge is not None:
            params['judge'] = judge
        
        params.update(kwargs)
        return self.client.paginate(self.endpoint, params=params)

File: php/src/CourtListener/Api/Investments.php
<?php

namespace CourtListener\Api;

use CourtListener\CourtListenerClient;
use CourtListener\Exceptions\CourtListenerException;

/**
 * Investments API endpoint
 */
class Investments extends BaseApi
{
    protected string $endpoint = 'investments/';

    /**
     * Constructor
     *
     * @param CourtListenerClient $client
     */
    public function __construct(CourtListenerClient $client)
    {
        parent::__construct($client);
    }

    /**
     * List investments with advanced filtering
     *
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function listInvestments(array $params = []) {
        return $this->list($params);
    }

    /**
     * Get a specific investments by ID
     *
     * @param int|string $id Investments ID
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function getInvestments(array $params = []) {
        return $this->get($id, $params);
    }

    /**
     * Search investments
     *
     * @param array $params Search parameters
     * @return array
     * @throws CourtListenerException
     */
    public function searchInvestments(array $params = []) {
        return $this->search($params);
    }
}
